### PR TITLE
Math magicians: Events

### PR DESCRIPTION
**Pull request by harlexkhal**
- [x] Copied the files ` Calculate.js` and `Operate.js`  into a logic/ directory in your project.
- [x] Analyzed the files ` Calculate.js` and ` Operate.js` and understood what they did.
- [x] Imported the files in your Calculator component.
- [x] Implemented the event handlers I needed to use the math logic from `Calculate.js`  and `Operate.js`  in your React component.
- [x] Practiced what I've learned from state and props, passing props to child components, using lifecycle methods, and lifting the state up.
- [x] Tested app in the browser. to make sure it performed all the math operations.
- [x] Launched to GitHub pages [here](https://harlexkhal.github.io/Mathemagician/)

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
-        <Button ctrls={['AC', '+/-', '%', '÷']} last={false} />
-        <Button ctrls={['7', '8', '9', 'X']} last={false} />
-        <Button ctrls={['4', '5', '6', '-']} last={false} />
-        <Button ctrls={['1', '2', '3', '+']} last={false} />
-        <Button ctrls={['0', '.', '=']} last />
-      </div>
-    );
-  }
-}
+const Calculator = () => {
```
Comment (Meltrust) [inline src/components/Calculator.js:6]: According to the project requirements:

image

Therefore:

- [ ] Please, kindly use lifecycle methods on your project, like `componentDidMout` for instance.

That way, we will meet this particular requirement.

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @harlexkhal,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice styling ✔️ 

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (harlexkhal) [reply src/components/Calculator.js]: Gotcha 💯

**Review by dasileker — APPROVED** [review]
### **Hi @harlexkhal,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 







Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by harlexkhal** [comment]
Thanks for the review 😃

**Pull request closed without merging**